src/osd/PGLog.cc

```diff
@@ -131,10 +131,8 @@ void PGLog::IndexedLog::trim(
     }
   }
 
-  while (!dups.empty()) {
+  while (!dups.empty() && dups.size() > cct->_conf->osd_pg_log_dups_tracked) {
```

Comment:
nit: there seem no need for `!dups.empty()` check now, ` dups.size() > cct->_conf->osd_pg_log_dups_tracked` would be enough.

Reply:
@trociny agree, done